test/membrane_rtc_engine/hls_endpoint_test.exs

```diff
@@ -166,15 +167,70 @@ defmodule Membrane.RTC.HLSEndpointTest do
       assert_receive({:cleanup, _cleanup_function, ^stream_id})
       refute_received({:cleanup, _cleanup_function, ^stream_id})
     end
+
+    test "check if number of headers is reduced to 1 when resolution is not stable", %{
+      rtc_engine: rtc_engine
+    } do
+      file_endpoint_id = "transcoding-file-endpoint-id"
+
+      file_name = "sample.h264"
+      file_path = Path.join(@fixtures_dir, file_name)
+
+      hls_endpoint_id = "hls-endpoint"
+
+      track_id = "test-track-id"
+      stream_id = "transcoding-test-stream"
+
+      hls_endpoint = create_hls_endpoint(rtc_engine, true)
+      :ok = Engine.add_endpoint(rtc_engine, hls_endpoint, endpoint_id: hls_endpoint_id)
+
+      file_endpoint =
+        create_video_file_endpoint(rtc_engine, file_path, stream_id, file_endpoint_id, track_id)
+
+      :ok = Engine.add_endpoint(rtc_engine, file_endpoint, endpoint_id: file_endpoint_id)
+
+      assert_receive %Message.MediaEvent{rtc_engine: ^rtc_engine, to: :broadcast, data: data}
+
+      assert %{
+               "type" => "tracksAdded",
+               "data" => %{
+                 "trackIdToMetadata" => %{track_id => nil},
+                 "peerId" => file_endpoint_id
+               }
+             } == Jason.decode!(data)
+
+      Engine.message_endpoint(rtc_engine, file_endpoint_id, :start)
+
+      assert_receive({:playlist_playable, :video, ^stream_id, ^file_endpoint_id}, 40_000)
+
+      Process.sleep(25_000)
+
+      Engine.remove_endpoint(rtc_engine, file_endpoint_id)
+
+      output_dir = Path.join([@output_dir, stream_id])
+      directory_files = File.ls!(output_dir)
+
+      # if number of header files is greater than 1, transcoding is not working properly
+      assert Enum.count(directory_files, fn file -> match?("video_header" <> _rest, file) end) ==
+               1
+
+      assert_receive {:cleanup, _cleanup_function, ^stream_id}
+
+      File.rm_rf!(@output_dir)
```

Comment:
You can use `@tag :tmp_dir` instead https://hexdocs.pm/ex_unit/ExUnit.Case.html#module-tmp-dir

Reply:
Fixed, also applied to other tests in `hls_endpoint_test.exs`.

Reply:
Great, thanks!